src/content/mod_configuration.cpp

```diff
@@ -196,7 +196,7 @@ void ModConfiguration::addModsFromConfig(
 	}
 }
 
-void ModConfiguration::checkConflictsAndDeps()
+void ModConfiguration::checkConflictsAndDeps(const std::string &first_mod, const std::string &last_mod)
```

Comment:
I think using empty strings for absent first / last mod is a bit dirty. Why not use `std::optional<std::string>` here to have some type safety?

Reply:
> shouldn't the `ModConfiguration` class store these actually? it seems to be responsible for sorting mods

Seems sensible. Might also simplify the code a bit by shortening parameter lists.
(Still the types should probably be `std::optional<std::string>`.)

Reply:
Now first and last mod are astored in class. No additional parameters.